app.js

```diff
@@ -0,0 +1,912 @@
+                console.log("cancel");
+                $mdDialog.cancel();
+            };
+
+            $scope.answer = function(answer) {
+                console.log("answer");
+                $mdDialog.hide(answer);
+            };
+        }
+
+    });
+
+
+(function () {
+    'use strict';
+    angular
+        .module('MyApp')
+        .controller('CustomInputCtrl', chipCtrl);
+
+    function chipCtrl ($timeout, $q, $scope, $mdDialog) {
+        console.log("chipCtrl fired");
+
+        var self = this;
+
+        /* Ng Material Chips directive properties */
+        self.readonly = false;
+        self.autocompleteDemoRequireMatch = false;
+        self.removable = false;
+
+        self.selectChip = function(chip) {
+            // console.log(chip);
+            var selectedChips = self.selectedChips;
+            if (selectedChips.indexOf(chip) == -1) {
+                selectedChips.push(chip);
+            } else {
+                selectedChips.pop(chip);
+            }
+
+            // console.log(selectedChips.indexOf(chip));
+        }
+
+        self.checkChip = function(chip) {
+            var checkChip = chip;
+            console.log(checkChip);
+        }
+
+        /**
+         * Return the proper object when the append is called.
+         */
+        function transformChip(chip) {
+            // If it is an object, it's already a known chip
+            if (angular.isObject(chip)) {
+                return chip;
+            }
+            // Otherwise, create a new one
+            return { name: chip }
+        }
+
+        /* SEARCH & Filtering */
+
+        // Array of Selected Chips within Modal
+        self.selectedChips = [];
+        // Selected item from autocomplete init
+        self.selectedItem = null;
+        // Current search text within any autocompkete
+        self.searchText = null;
+        // Array of Suggested items that have been selected
+        // self.selectSuggest = [];
+
+        // Define query - NB: Seems easier to define seperate queries if intending multiple search modals, due to changing fields on chips.
+        self.querySearchThing = querySearchThing;
+
+//   *****************
+        /* 1. THING. This is one chip-dialog-search-module. It is intended that you could copy and refactor this section for multiple different searches, sharing the above code between them.*/
+        /* Some functions are not easily made DRY, mainly because of custom fields on chips */
+
+        self.Thing = loadThing();
+
+        self.transformThingChip = transformThingChip;
+        function transformThingChip(chip) {
+            // If it is an object, it's already a known chip
+            if (angular.isObject(chip)) {
+                return chip;
+            }
+            // Otherwise, create a new one
+            return { thing: chip }
+        }
+        // Define array of "things" into a function to load at once
+        function loadThing() {
+
+            var setarray = [
```

Comment:
Having this data here makes this file much harder to read.

You could move the data structure to another file and have it load separately / first